From b5d1f34e9e056d3637bd06dcaff9a6383131b3d7 Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Fri, 20 Dec 2019 11:30:05 +0100
Subject: [PATCH 033/102] lima: Fix dump file creation

Otherwise lima_dump_file_next() always opens a new file and creates the
dumps regardless of what the environment variables say.

Fixes d71cd245d74 ('lima: Rotate dump files after each finished pp frame')

Reviewed-by: Vasily Khoruzhick <anarsoul@gmail.com>
Signed-off-by: Andreas Baierl <ichgeh@imkreisrum.de>
Tested-by: Marge Bot <https://gitlab.freedesktop.org/mesa/mesa/merge_requests/3179>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/merge_requests/3179>
---
 src/gallium/drivers/lima/lima_util.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/gallium/drivers/lima/lima_util.c b/src/gallium/drivers/lima/lima_util.c
index 65e3114ed5f..45f2b27b400 100644
--- a/src/gallium/drivers/lima/lima_util.c
+++ b/src/gallium/drivers/lima/lima_util.c
@@ -130,9 +130,11 @@ lima_dump_file_close(void)
 void
 lima_dump_file_next(void)
 {
-   lima_dump_file_close();
-   lima_dump_frame_count++;
-   lima_dump_file_open();
+   if (lima_dump_command_stream) {
+      lima_dump_file_close();
+      lima_dump_frame_count++;
+      lima_dump_file_open();
+   }
 }
 
 void
-- 
2.24.1

